websockify/websocketproxy.py

```diff
@@ -457,44 +464,53 @@ def websockify_init():
 
     del opts.target_cfg
 
-    # Sanity checks
-    if len(args) < 2 and not (opts.token_plugin or opts.unix_target):
-        parser.error("Too few arguments")
     if sys.argv.count('--'):
         opts.wrap_cmd = args[1:]
     else:
         opts.wrap_cmd = None
-        if len(args) > 2:
-            parser.error("Too many arguments")
 
     if not websockifyserver.ssl and opts.ssl_target:
         parser.error("SSL target requested and Python SSL module not loaded.");
 
     if opts.ssl_only and not os.path.exists(opts.cert):
         parser.error("SSL only and %s not found" % opts.cert)
 
-    # Parse host:port and convert ports to numbers
-    if args[0].count(':') > 0:
-        opts.listen_host, opts.listen_port = args[0].rsplit(':', 1)
-        opts.listen_host = opts.listen_host.strip('[]')
+    if opts.inetd:
+        opts.listen_fd = sys.stdin.fileno()
     else:
-        opts.listen_host, opts.listen_port = '', args[0]
+        if len(args) < 1:
+            parser.error("Too few arguments")
+        arg = args.pop(0)
+        # Parse host:port and convert ports to numbers
+        if arg.count(':') > 0:
+            opts.listen_host, opts.listen_port = arg.rsplit(':', 1)
+            opts.listen_host = opts.listen_host.strip('[]')
+        else:
+            opts.listen_host, opts.listen_port = '', arg
 
-    try:    opts.listen_port = int(opts.listen_port)
-    except: parser.error("Error parsing listen port")
+        try:    opts.listen_port = int(opts.listen_port)
+        except: parser.error("Error parsing listen port")
+
+    del opts.inetd
```

Comment:
Because the `opts` are eventually used as parameters of `WebSockifyServer` constructor and it would fail because it doesn't know the `inetd` parameter. This deletion follows the same pattern as for example `opts.target_cfg` few lines above.